File: VB/Extensions/MyColumnTemplateSelector.vb
Imports DXGridSample.ViewModels
Imports System.Windows
Imports System.Windows.Controls

Namespace DXGridSample.Extensions
    Public Class MyColumnTemplateSelector
        Inherits DataTemplateSelector

        Public Property template1() As DataTemplate
        Public Property template2() As DataTemplate
        Public Overrides Function SelectTemplate(ByVal item As Object, ByVal container As DependencyObject) As DataTemplate
            Dim ci As ColumnViewModel = TryCast(item, ColumnViewModel)
            If ci IsNot Nothing Then
                If ci.ColumnFieldName = "Name" Then
                    Return template2
                End If
                Return template1
            End If
            Return MyBase.SelectTemplate(item, container)
        End Function
    End Class
End Namespace

File: CS/Extensions/MyColumnTemplateSelector.cs
using DXGridSample.ViewModels;
using System.Windows;
using System.Windows.Controls;

namespace DXGridSample.Extensions {
    public class MyColumnTemplateSelector : DataTemplateSelector {
        public DataTemplate template1 { get; set; }
        public DataTemplate template2 { get; set; }
        public override DataTemplate SelectTemplate(object item, DependencyObject container) {
            ColumnViewModel ci = item as ColumnViewModel;
            if (ci != null) {
                if (ci.ColumnFieldName == "Name") return template2;
                return template1;
            }        
            return base.SelectTemplate(item, container);
        }
    }
}